source/common/grpc/async_client_impl.cc

```diff
@@ -14,6 +14,19 @@
 namespace Envoy {
 namespace Grpc {
 
+namespace {
+std::string enhancedGrpcMessage(const std::string& original_message,
+                                const Http::AsyncClient::Stream* stream) {
+  const auto& http_response_code_details = (stream && stream->streamInfo().responseCodeDetails())
+                                               ? *stream->streamInfo().responseCodeDetails()
+                                               : EMPTY_STRING;
+  return original_message.empty() ? http_response_code_details
```

Comment:
Can we change the logic like below:
if original_message is empty, then using http_response_code_details. 
otherwise, using absl::StrCat(original_message, "{", http_response_code_details, "}");   And http_response_code_details can be empty.

Reply:
I don't want `{}` present if there is no HTTP RCD.